### PR TITLE
♻ Create training script, add flake.nix

### PR DESCRIPTION
**Pull request by dpaetzel**

### CODE REVIEW

File: train_model.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# !/usr/bin/env python
+
+import numpy as np
+import tensorflow as tf
+from dataset import get_dataset
+import click
+
+
+@click.command()
```
Comment (dpaetzel) [inline train_model.py:9]: I made this a script so we can do

```bash
git clone https://github.com/dpaetzel/falken-net
cd falken-net
python train_model.py --learning-rate=0.00001
```

File: train_model.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    help=
+    "Number of epochs to train for during fine tuning (requires --fine-tune)",
+    default=10,
+    type=int)
+@click.option(
+    "--fine-tune",
+    help=
+    "Whether to perform fine tuning (i.e. train the mobile net weights as well)",
+    default=False)
+@click.option("--softmax",
```
Comment (dpaetzel) [inline train_model.py:28]: I added an option to disable the softmax layer.
